### ISSUE TITLE
Use OSM widths when properly specified in detail

### ISSUE THREAD

**Issue by mvl22** [cyipt/cyipt-bigdata#1]
The recent e-mail correspondence from a stakeholder gave some example roads that had very constrained width. These are probably on the more unusual side.

It would be sensible for the importer to use specific widths when they exist in OSM (with both road and pavement widths taken into account). This way, more unusual cases can be added into the raw OSM data and the application can then pick these up.

It would be useful to use the supplied examples and ask the stakeholder to give us measurements, which we can then add into the OSM and use as test-cases.

**Comment by Robinlovelace** [comment]
Agreed. I think in the long-run the CyIPT web app should encourage people to add info to OSM directly and respond to these changes. As discussed with Roger Geffen, this could lead CyIPT to provide LoS-type outputs in response to user input *on the OSM network* which is a much more useful place to put infrastructure details than in some side project. Are there any fields that OSM does not support.

I'm not quite sure how it handles road widths but imagine there's a field that could overide the automated width calculation. Is that right?

**Comment by mvl22** [comment]
> OSM network which is a much more useful place to put infrastructure details than in some side project

+1.

OSM is indeed the correct place, architecturally-speaking, where deals about road widths should be put.

> I'm not quite sure how it handles road widths but imagine there's a field that could overide the automated width calculation. Is that right?

Yes, this is what I'm suggesting that we add code support for.

People can add tags relating to road widths (which is not straight-forward - we would need to give clear instructions, and this is a moving target anyway due to improvements in editor support), and then code in CyIPT can pick those up and use these direct values instead of whatever we've attempted to derive from another source.

**Comment by Robinlovelace** [comment]
OK great. This links to wider plans that have discussed. Will fill you in via audio soon.
